src/NUnitFramework/framework/Internal/Commands/WarnCommand.cs

```diff
@@ -0,0 +1,59 @@
+// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using NUnit.Framework.Interfaces;
+
+namespace NUnit.Framework.Internal.Commands
+{
+    /// <summary>
+    /// This innerCommand will add a warning to the test result
+    /// </summary>
+    public class WarnCommand : TestCommand
```

Comment:
Clever! I see pros and cons to this approach, which we should discuss.

Reply:
😨 ... This was an interesting decision... I saw the `SkipCommand` but there was nothing else that did what I needed. I looked at other ways to add a warning but couldn't see any other examples in the code to template. I didn't want to pollute code with a check (if `(method.GetAttributes<IRepeatTest>(true).Length > 1)`) like this anywhere else and it seemed to nicely where it was at the end of the command chain.

It didn't break any current tests so I couldn't see (with my limited knowledge) any downside but I was wondering if I would be asked to explain myself 😉

Reply:
😨 ... This was an interesting decision... I saw the `SkipCommand` but there was nothing else that did what I needed. I looked at other ways to add a warning but couldn't see any other examples in the code to template. I didn't want to pollute code with a check ('if (method.GetAttributes<IRepeatTest>(true).Length > 1)`) like this anywhere else and it seemed to nicely where it was at the end of the command chain.

It didn't break any current tests so I couldn't see (with my limited knowledge) any downside but I was wondering if I would be asked to explain myself 😉